### DISCUSSION TITLE
Twitch provider callback accesstoken

### DISCUSSION THREAD

**Discussion by stevealexandre** [nextauthjs/next-auth#6000] [Help]
Hey everyone,

I'm trying since a few days to get the twitch access token after logon by following the getting started doc.  The authentication is working but for moment I can't still get my access token to use it for the api call to twitch.
With the debug mode enabled, I can see it present in the account object but trying to defined the callback to store it and pass it to the session seems not working. Also trying to console log in the jwt callback does not display any log in my console. It's look like it's not passing in it and so in my session I have only have my user(name, email, image) and expires data.
Can someone tell me what I'm doing wrong to fix it ?

My NextAuth api file under api/auth :
```
import NextAuth from "next-auth";
import TwitchProvider from "next-auth/providers/twitch";

export const authOptions = {
  // Configure one or more authentication providers
  providers: [
    TwitchProvider({
      clientId: process.env.TWITCH_ID,
      clientSecret: process.env.TWITCH_SECRET,
      authorization: { params: { scope: "openid user:read:email moderation:read moderator:manage:banned_users" } },    }),
  ],
  debug: false,
  callback: {
    async jwt({ token, account }) {
      // Persist the OAuth access_token to the token right after signin
      if (account) {
        token.accessToken = account.access_token
      }
      console.log(token)
      return token
    },
    async session({ session, token, user }) {
      // Send properties to the client, like an access_token from a provider.
      session.accessToken = token.accessToken
      return session
    }
  },
};

export default NextAuth(authOptions);
```

I've done many search before creating this topic and found nothing helping me after many tried so sorry if it's really a stupid mistake to fix it.
Thanks for your help.

**Comment by pierretusseau** [comment]
According to this page : https://next-auth.js.org/configuration/nextjs#in-app-router

You need to pass `authOptions` as a parametter for `getServerSession()` from

For Page Router
```import { authOptions } from "pages/api/auth/[...nextauth]"```

For App Router
```import { authOptions } from "src/app/api/auth/[...nextauth]"```

to get the accessToken you defined in those very options as callback for the session.

It took me some time to realize it was my problem, since this post is very visible on Google for this specific problem I hope it can solve the issue for you guys.
